server/remote_cache/content_addressable_storage_server/content_addressable_storage_server_test.go

```diff
@@ -48,6 +48,70 @@ func (e *evilCache) GetMulti(ctx context.Context, digests []*repb.Digest) (map[*
 	return rsp, err
 }
 
+func TestBatchUpdateBlobs(t *testing.T) {
+	ctx := context.Background()
+	te := testenv.GetTestEnv(t)
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, te)
+	if err != nil {
+		t.Errorf("error attaching user prefix: %v", err)
+	}
+
+	clientConn := runCASServer(ctx, te, t)
+	casClient := repb.NewContentAddressableStorageClient(clientConn)
+
+	req := &repb.BatchUpdateBlobsRequest{}
+	for i := 0; i < 100; i++ {
+		d, buf := testdigest.NewRandomDigestBuf(t, 100)
+		req.Requests = append(req.Requests, &repb.BatchUpdateBlobsRequest_Request{
+			Digest: d,
+			Data:   buf,
+		})
+	}
+	rsp, err := casClient.BatchUpdateBlobs(ctx, req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, 100, len(rsp.GetResponses()))
+	for _, singleRsp := range rsp.GetResponses() {
+		assert.Equal(t, int32(gcodes.OK), singleRsp.GetStatus().GetCode())
+	}
+}
+
+func TestBatchUpdateRejectCorruptBlobs(t *testing.T) {
+	ctx := context.Background()
+	te := testenv.GetTestEnv(t)
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, te)
+	if err != nil {
+		t.Errorf("error attaching user prefix: %v", err)
+	}
+
+	clientConn := runCASServer(ctx, te, t)
+	casClient := repb.NewContentAddressableStorageClient(clientConn)
+
+	req := &repb.BatchUpdateBlobsRequest{}
+	d, buf := testdigest.NewRandomDigestBuf(t, 100)
+	buf[0] = ^buf[0] // corrupt the data in buf
+	req.Requests = append(req.Requests, &repb.BatchUpdateBlobsRequest_Request{
```

Comment:
Maybe worth adding a third request that is not corrupted?

Reply:
Done.